### PR TITLE
Adds Python 3.8; build for Jammy

### PR DESCRIPTION
**Pull request by bobokun**
Splits the commits into multiple commits for easier review.

This reverts PR https://github.com/Yelp/Tron/pull/879 and fixes dependencies and tests

### CODE REVIEW

**Review by nemacysts — APPROVED** [review]

**Comment by ahaswell** [comment]
Could you check if 3.8 is an easy bump as 3.7 support ends in June.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -49,5 +49,5 @@ repos:
     rev: 22.3.0
     hooks:
       - id: black
-        language_version: python3.7
```
Comment (nemacysts) [inline .pre-commit-config.yaml]: nit: we can probably get rid of this - this should be configured by the default_language_version at the top of this file

File: bin/tronctl [hunk 2]
```diff
@@ -103,7 +103,7 @@ def parse_date(date_string):
 def parse_cli():
     parser = cmd_utils.build_option_parser()
     subparsers = parser.add_subparsers(dest="command", title="commands", help="Tronctl command to run")
-    subparsers.required = True  # add_subparsers only supports required arg from py37
+    subparsers.required = True  # add_subparsers only supports required arg from py38
```
Comment (nemacysts) [inline bin/tronctl:106]:
```python
subparsers = parser.add_subparsers(dest="command", title="commands", help="Tronctl command to run", required=True)
``` 
should work here now that we're gonna be running on 3.7+ :)

(this comment was referring to in `required - Whether or not a subcommand must be provided, by default False (added in 3.7)` https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_subparsers

File: setup.py [hunk 3]
```diff
@@ -19,7 +19,7 @@
     classifiers=[
         "Programming Language :: Python",
         "Programming Language :: Python :: 3.6",
```
Comment (nemacysts) [inline setup.py:21]: whoops, missed this before - can we also remove this line?

**Review by nemacysts — APPROVED** [review]
thank you! i have some minor nits/suggestions, but none are blocking

**Comment by nemacysts** [comment]
ah, also need to update https://github.com/Yelp/Tron/blob/7de44d2ae50f87ea75519cc8c82ffb187b3157ee/itest.sh#L22

**Review by nemacysts — APPROVED** [review]

File: yelp_package/bionic/Dockerfile [hunk 4]
```diff
@@ -18,9 +18,10 @@ RUN apt-get -q update && \
         libyaml-dev \
         libssl-dev \
         libffi-dev \
-        python3.6-dev \
+        python3.8-dev \
         python3-pip \
         python-tox \
+        rust-all \
```
Comment (EvanKrall) [inline yelp_package/bionic/Dockerfile:24]: Rust?

**Review by EvanKrall — APPROVED** [review]

**Pull request merged**